pkgs/development/python-modules/minikanren/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, toolz
+, cons
+, multipledispatch
+, etuples
+, logical-unification
+, pydocstyle
+, black
+, pylint
+, pytest
+, pytest-cov
+, pytest-html
+}:
+
+buildPythonPackage rec {
+  pname = "minikanren";
+  version = "1.0.3";
+
+  src = fetchFromGitHub {
+    owner = "pythological";
+    repo = "kanren";
+    rev = "5aa9b1734cbb3fe072a7c72b46e1b72a174d28ac";
+    sha256 = "sha256-daAtREgm91634Q0mc0/WZivDiyZHC7TIRoGRo8hMnGE=";
+  };
+
+  propagatedBuildInputs = [
+    toolz
+    cons
+    multipledispatch
+    etuples
+    logical-unification
+  ];
+
+  checkInputs = [
+    pydocstyle
+    black
+    pylint
+    pytest
+    pytest-cov
+    pytest-html
+  ];
+
+  checkPhase = ''
+    make lint
+    make test
+  '';
```

Comment:
We don't want to run lint tests. They only break over time.

and instead of running make test we want to run pytestCheckHook with the additional flags:
``--html=testing-report.html --self-contained-html``.